### PR TITLE
Skip install media in the list of available disks

### PR DESCRIPTION
**Pull request by pyhalov**

### CODE REVIEW

File: usr/src/lib/libtd/td_dd.c [hunk 1]
```diff
@@ -1103,6 +1115,73 @@ ddm_drive_is_floppy(ddm_handle_t d)
...
+ddm_drive_is_install_media(ddm_handle_t d)
+{
+	int		errn;
+	nvlist_t	*dz;
+	char		*dn, *device_path;
+	boolean_t	drive_is_install_media = B_FALSE;
+	struct mnttab 	mp, search_mp;
+	FILE *mnttab;
+
+	dn=ddm_drive_get_name(d);
```
Comment (tsoome) [inline usr/src/lib/libtd/td_dd.c]: spaces around =?

File: usr/src/lib/libtd/td_dd.c [hunk 2]
```diff
@@ -1103,6 +1115,73 @@ ddm_drive_is_floppy(ddm_handle_t d)
...
+ */
+static boolean_t
+ddm_drive_is_install_media(ddm_handle_t d)
+{
+	int		errn;
+	nvlist_t	*dz;
+	char		*dn, *device_path;
+	boolean_t	drive_is_install_media = B_FALSE;
+	struct mnttab 	mp, search_mp;
+	FILE *mnttab;
```
Comment (tsoome) [inline usr/src/lib/libtd/td_dd.c:1138]: tab to align with other variables

File: usr/src/lib/libtd/td_dd.c [hunk 3]
```diff
@@ -1103,6 +1115,73 @@ ddm_drive_is_floppy(ddm_handle_t d)
...
+                return (1);
+
+	/*
+         * If device path can't be obtained, filter the drive out as well
+         */
+	if ((device_path = ddm_get_device_path_from_ctd_name(dn, ',', B_FALSE)) == NULL) {
+                DDM_DEBUG(DDM_DBGLVL_INFO, "Could not obtain physical device "
+                    "path for disk %s\n", dn);
+		dm_free_name(dn);
+		return 1;
```
Comment (tsoome) [inline usr/src/lib/libtd/td_dd.c]: return (1);

File: usr/src/lib/libtd/td_dd.c [hunk 4]
```diff
@@ -1103,6 +1115,73 @@ ddm_drive_is_floppy(ddm_handle_t d)
...
+        }
+        DDM_DEBUG(DDM_DBGLVL_INFO, "Physical device path for disk %s: "
+		"%s\n", dn, device_path);
+
+	mnttab=fopen("/etc/mnttab","r");
```
Comment (tsoome) [inline usr/src/lib/libtd/td_dd.c]: spaces around =

File: usr/src/lib/libtd/td_dd.c [hunk 5]
```diff
@@ -1103,6 +1115,73 @@ ddm_drive_is_floppy(ddm_handle_t d)
...
+	/*
+	 * If we can't open mnttab, filter the drive out as well
+	 */
+	if (mnttab == NULL) {
+		DDM_DEBUG(DDM_DBGLVL_ERROR, "Couldn't open mnttab. Something is terribly wrong. ");
+		dm_free_name(dn);
+		free(device_path);
+		return 1;
```
Comment (tsoome) [inline usr/src/lib/libtd/td_dd.c]: return (1);

File: usr/src/lib/libtd/td_dd.c [hunk 6]
```diff
@@ -1103,6 +1115,73 @@ ddm_drive_is_floppy(ddm_handle_t d)
...
+	}
+
+	bzero(&search_mp, sizeof(struct mnttab));
+	search_mp.mnt_special = device_path;
+	search_mp.mnt_mountp = DDM_INSTALL_MEDIA_MOUNTPOINT;
+	if(!getmntany(mnttab, &mp, &search_mp)) {
```
Comment (tsoome) [inline usr/src/lib/libtd/td_dd.c]: space after if

**Review by tsoome — CHANGES_REQUESTED** [review]

File: usr/src/lib/libtd/td_dd.c (hunk 4 above)
Comment (xen0l) [inline usr/src/lib/libtd/td_dd.c]: mnttab=fopen("/etc/mnttab","r"); -> mnttab = fopen("/etc/mnttab","r");

**Review by xen0l — CHANGES_REQUESTED** [review]

File: usr/src/lib/libtd/td_dd.c (hunk 4 above)
Comment (pyhalov) [reply usr/src/lib/libtd/td_dd.c]: fixed

**Pull request merged**